src/Microsoft.OData.Core/ODataWriterCore.cs

```diff
@@ -2339,7 +2339,7 @@ private void EnterScope(WriterState newState, ODataItem item)
 
                             if (resourceTypeCast != null)
                             {
-                                odataPath.Add(resourceTypeCast);
+                                odataPath = odataPath.AppendSegment(resourceTypeCast);

```

Comment:
>odataPath = odataPath.AppendSegment(resourceTypeCast); [](start = 32, length = 54)

what i mean is that:

If it's "append", we don't need return a new ODataPath.

Can it be directly "  odataPath.AppendSegment(resourceTypeCast);"

Reply:
See other comment.  I just kept this consistent with the existing naming convention, rather than changing it everywhere, which I could do.

---
In reply to: [539590070](https://github.com/OData/odata.net/pull/1930#discussion_r539590070) [](ancestors = 539590070)

Reply:
Renamed all of these to Add*.

---
In reply to: [542970463](https://github.com/OData/odata.net/pull/1930#discussion_r542970463) [](ancestors = 542970463,539590070)